data/test/scenarios/wml_tests/ScenarioWML/EventWML/events-test_filters.cfg

```diff
@@ -294,20 +294,175 @@
+                            [/filter_weapon]
+                        [/filter_opponent]
+                    [/poison]
+                [/abilities]
+            [/effect]
+            [filter]
+                id=bob
+            [/filter]
+        [/object]
+        [object]
+            silent=yes
+            [effect]
+                apply_to=new_ability
+                [abilities]
+                    [drains]
+                        id=ability_drain_blade
+                        [filter_opponent]
+                            [filter_weapon]
+                                {FILTER}
+                            [/filter_weapon]
+                        [/filter_opponent]
+                    [/drains]
+                [/abilities]
+            [/effect]
+            [filter]
+                id=alice
+            [/filter]
+        [/object]
+        [modify_unit]
+            [filter]
+            [/filter]
+            # Make sure they don't die during the attacks
+            [status]
+                invulnerable=yes
+            [/status]
+        [/modify_unit]
+        {VARIABLE triggers 0}
+        {VARIABLE triggers_on_attack 0}
+        {VARIABLE triggers_on_defense 0}
+        [do_command]
+            [move]
+                x=7,13
+                y=3,4
+            [/move]
+            [attack]
+                [source]
+                    x,y=13,4
+                [/source]
+                [destination]
+                    x,y=13,3
+                [/destination]
+            [/attack]
+        [/do_command]
+        [end_turn][/end_turn]
+    [/event]
+    [event]
+        name=side 2 turn
+        [do_command]
+            [attack]
+                [source]
+                    x,y=13,3
+                [/source]
+                [destination]
+                    x,y=13,4
+                [/destination]
+            [/attack]
+        [/do_command]
+        [end_turn][/end_turn]
+    [/event]
+    [event]
+        name=attack
+        first_time_only=no
+        [filter_attack]
+            special_id_active=ability_poison_blade
+        [/filter_attack]
+        {ASSERT ({VARIABLE_CONDITIONAL side_number equals 2})}
+        {VARIABLE_OP triggers_on_attack add 1}
+    [/event]
+    [event]
+        name=attack
+        first_time_only=no
+        [filter_second_attack]
+            special_id_active=ability_poison_blade
+        [/filter_second_attack]
+        {ASSERT ({VARIABLE_CONDITIONAL side_number equals 1})}
+        {VARIABLE_OP triggers_on_defense add 1}
+    [/event]
+#enddef
+
 #####
 # API(s) being tested: [event][filter_attack],[event][filter_second_attack]
 ##
 # Actions:
 # Define events filtering based on weapon specials.
 # Give Bob a poison special iff the opponent has a specific drain special.
-# Give Alice a drain special iff the opponent has a specific poison special.
+# Give Alice that drain special iff the opponent has a blade attack.
```

Comment:
'Give Alice **a** drain special iff the opponent has a blade attack.'

(Je connais votre langue maternelle n'est pas Anglais...)

As an aside: what is the significance of the blade attack being tested here? Was the blade attack a cause of the infinite recursion previously? (I'm not familiar with the details of #9202)

Reply:
No, tis test a case without recursion else we would have a BROKE STRICT.